lib/ansible/galaxy/collection/__init__.py

```diff
@@ -769,6 +770,19 @@ def install_collections(
                     "Skipping signature verification."
                 )
 
+            if concrete_coll_pin.type == 'galaxy':
+                signatures = concrete_coll_pin.src.get_collection_signatures(
+                    concrete_coll_pin.namespace, concrete_coll_pin.name, concrete_coll_pin.ver
+                )
+
+                concrete_coll_pin = Candidate(
+                    concrete_coll_pin.fqcn,
+                    concrete_coll_pin.ver,
+                    concrete_coll_pin.src,
+                    concrete_coll_pin.type,
+                    frozenset([*concrete_coll_pin.signatures, *signatures])
+                )
```

Comment:
This calls for implementing a conversion helper method in the candidate. Smth like this:
```suggestion
                concrete_coll_pin = concrete_coll_pin.with_signatures(*concrete_coll_pin.signatures, *signatures)
```